librz/core/agraph.c

```diff
@@ -3705,18 +3732,11 @@ RZ_API RzANode *rz_agraph_add_node_with_color(const RzAGraph *g, const char *tit
 	if (res) {
 		return res;
 	}
-	res = RZ_NEW0(RzANode);
+	res = agraph_node_new(title, body);
 	if (!res) {
 		return NULL;
 	}
 
-	res->title = title ? rz_str_trunc_ellipsis(title, 255) : strdup("");
-	res->body = body ? strdup(body) : strdup("");
-	res->layer = -1;
-	res->pos_in_layer = -1;
-	res->is_dummy = false;
-	res->is_reversed = false;
-	res->klass = -1;
 	res->difftype = color;
 	res->gnode = rz_graph_add_node(g->graph, res);
 	sdb_num_set(g->nodes, res->title, (ut64)(size_t)res, 0);
```

Comment:
I'm wondering if this should become:
```suggestion
	if (title) {
		sdb_num_set(g->nodes, res->title, (ut64)(size_t)res, 0);
```
(without the `if (res->title)`).

At that point, you could just use `rz_agraph_add_node_with_color` instead of extracting the first part of the function and then duplicating the `res->gnode =` part in https://github.com/rizinorg/rizin/pull/715/files#diff-9ed74cc0401d4466ebfa148ce6f4a0a1e88c04627dd4c998443eda8f88134280R755